components/nexus-core/src/main/java/org/sonatype/nexus/proxy/item/ChecksumReconciler.java

```diff
@@ -114,7 +129,46 @@ void reconcileItemChecksum(final Repository repo, final StorageFileItem item) th
         finally {
           uid.getLock().unlock();
         }
+
+        final String itemPath = item.getPath();
+        if (!itemPath.endsWith(".sha1") && !itemPath.endsWith(".md5")) {
```

Comment:
No - the code above is reconciling checksums stored in the item attributes, so for example a file called "foo.bar" will have checksums stored in the attribute system under "foo.bar. It may also have those same checksums persisted in local storage as "foo.bar.sha1" and/or "foo.bar.md5" but this is not always the case.